apps/site/components/Common/Search/index.tsx

```diff
@@ -1,61 +1,132 @@
+    .join(' > ')} — ${title}`;
 
-export const SearchButton: FC = () => {
-  const [isOpen, setIsOpen] = useState(false);
+const SearchButton: FC = () => {
+  const { resolvedTheme } = useTheme();
   const t = useTranslations();
-  const openSearchBox = () => setIsOpen(true);
-  const closeSearchBox = () => setIsOpen(false);
-
-  useKeyboardCommands(cmd => {
-    switch (cmd) {
-      case 'cmd-k':
-        openSearchBox();
-        break;
-      case 'escape':
-        closeSearchBox();
-        break;
-      default:
-    }
-  });
-
-  const { os } = useDetectOS();
-
-  const osCommandKey = os === 'MAC' ? '⌘' : 'Ctrl';
-  const isOSLoading = os === 'LOADING';
+  const locale = useLocale();
+  const colorScheme = resolvedTheme as 'light' | 'dark';
+  const router = useRouter();
+
+  const sourceMap = {
+    title: 'pageSectionTitle',
+    description: 'formattedPath',
+    path: 'path',
+  };
+
+  const resultMap = {
+    ...sourceMap,
+    description: ({ path, pageSectionTitle }: ResultMapDescription) =>
+      getFormattedPath(path, pageSectionTitle),
+    path: ({ path, siteSection }: ResultMapPath) =>
+      siteSection.toLowerCase() === 'docs' ? `/${path}` : `/${locale}/${path}`,
+    section: 'siteSection',
+  };
 
   return (
     <>
-      <button
-        type="button"
-        onClick={openSearchBox}
-        className={styles.searchButton}
+      <OramaSearchButton
+        style={{ flexGrow: 1 }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
         aria-label={t('components.search.searchBox.placeholder')}
       >
-        <MagnifyingGlassIcon className={styles.magnifyingGlassIcon} />
-
         {t('components.search.searchBox.placeholder')}
-        <kbd
-          title={`${osCommandKey} K`}
-          className={classNames(styles.shortcutIndicator, {
-            'opacity-0': isOSLoading,
-          })}
-        >
-          <abbr>{osCommandKey} K</abbr>
-        </kbd>
-      </button>
-
-      {isOpen ? <WithSearchBox onClose={closeSearchBox} /> : null}
+      </OramaSearchButton>
+
+      <OramaSearchBox
+        index={{ api_key: ORAMA_CLOUD_API_KEY, endpoint: ORAMA_CLOUD_ENDPOINT }}
+        colorScheme={colorScheme}
+        themeConfig={themeConfig}
+        sourceBaseUrl={BASE_URL}
+        sourcesMap={sourceMap}
+        resultMap={resultMap}
+        facetProperty="siteSection"
+        linksTarget="_self"
+        highlightTitle={{
+          caseSensitive: false,
+          HTMLTag: 'b',
+          CSSClass: 'font-bold',
+        }}
+        searchParams={DEFAULT_ORAMA_QUERY_PARAMS}
+        suggestions={DEFAULT_ORAMA_SUGGESTIONS}
+        chatMarkdownLinkHref={({ href }) => {
+          if (!href) return href;
+
+          const currentPageHost = `${window.location.protocol}//${window.location.host}`;
+          const docsBaseUrl = `${currentPageHost}/docs`;
+          const lowerCaseHref = href.toLowerCase();
+
+          // Keep original link if URl should not have locale
```

Comment:
Can you add the reasoning? This whole URL logic seems unnecessary, you can simply check if the `href`'s pathname startsWith docs. You can accomplish that by

`const { pathname }= new URL(href);`

Reply:
Hi! The problem with your approach is that you are assuming that LLM will always reply with URLs related under nodes domain. What if one asks: can you give me the documentation URL from some companies that uses NodeJs?

Does it make sense  to you?

Reply:
Why do we care about comparing content from other domains? That's a pretty easy, if the URL does not start with `https://nodejs.org` then simply return `href`

That can be 100% hardcoded there and it is the simplest yet best approach here; You can add inline documentary explaining why and that the href can be other domains but we only care about nodejs.org domain

Reply:
(Alternatively you can check against BASE_URL), the point is, these checks need to be deterministic. Using `window.*` is not deterministic and environment-specific.

Reply:
Hi, I believe we are not comparing content from other domains here. The only thing we're doing is to compare:
1) If domain is different
2) If domain is equal, but path starts with /docs/

Anyways, I have rewrote this part in order to use URL object and BASE URL. Would you mind to check it? Thanks!

Reply:
I'll re-read the code to better understand it them, so hold tight.